idd/CMakeLists.txt

```diff
@@ -1,21 +1,65 @@
-#install(FILES "V7-2-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-#install(FILES "V8-0-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-#install(FILES "V8-1-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-#install(FILES "V8-2-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-#install(FILES "V8-3-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-#install(FILES "V8-4-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-#install(FILES "V8-5-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-#install(FILES "V8-6-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-#install(FILES "V8-7-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-#install(FILES "V8-8-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-#install(FILES "V8-9-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-install(FILES "V9-0-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-install(FILES "V9-1-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-install(FILES "V9-2-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-install(FILES "V9-3-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-install(FILES "V9-4-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-install(FILES "V9-5-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-install(FILES "V9-6-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-install(FILES "V22-1-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-install(FILES "V22-2-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
-install(FILES "V23-1-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
+configure_file("Energy+.idd.in" "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/Energy+.idd")
+
+
+add_custom_command(
+  OUTPUT "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/Energy+.schema.epJSON"
+  COMMAND ${Python_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/schema/generate_epJSON_schema.py" "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}"
+  MAIN_DEPENDENCY "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/Energy+.idd"
+  DEPENDS
+    schema/generate_epJSON_schema.py
+    schema/idd_parser.py
+    schema/modify_schema.py
+  VERBATIM
+)
+
+add_executable(generate_embeddable_epJSON_schema embedded/generate_embeddable_epJSON_schema.cpp)
+target_link_libraries(generate_embeddable_epJSON_schema PRIVATE project_options project_fp_options project_warnings fmt::fmt)
+set_target_properties(generate_embeddable_epJSON_schema PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/scripts")
+# See https://en.cppreference.com/w/cpp/filesystem#Notes
+if ((CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1) OR
+    ( ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)))
+  target_link_libraries(generate_embeddable_epJSON_schema PRIVATE stdc++fs)
+endif()
```

Comment:
I guess I could `file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/embedded")` instead of relying on std::filesystem to create the directory if needed, but I think it's clearer this way, all logic is within the C++ file instead